### DISCUSSION TITLE
dashboard新增一条路由导致所有域名访问都出现问题

### DISCUSSION THREAD

**Discussion by Lewisyixin** [apache/apisix#7706] [Q&A]
是通过dashboard添加的，目前环境没动的情况下能稳定复现，但是没有好的办法把问题展现出来。
现象就是问题路由只要一点发布按钮，其他所有域名的访问全部出现问题，并且每次刷新所路由到的后端都不一样
![image](https://user-images.githubusercontent.com/17898688/184909195-584964e5-8153-4aa7-aabe-bb0567fd9804.png)
将路由下线并reload apisix后恢复，再次点上线能稳定复现问题，问题路由配置如下，apisix和etcd均无error日志，提issue感觉没有办法清晰的表达出问题的现象，所以想先了解下类似情况有没有什么能进一步自查的方法？

```
{
  "uri": "/log/api/json",
  "name": "xxxx",
  "methods": [
    "GET",
    "POST",
    "PUT",
    "DELETE",
    "PATCH",
    "HEAD",
    "OPTIONS",
    "CONNECT",
    "TRACE"
  ],
  "host": "xxxx.com",
  "upstream": {
    "nodes": [
      {
        "host": "xxxx11.com",
        "port": 443,
        "weight": 100
      }
    ],
    "timeout": {
      "connect": 6,
      "send": 6,
      "read": 6
    },
    "type": "roundrobin",
    "scheme": "https",
    "pass_host": "node",
    "keepalive_pool": {
      "idle_timeout": 60,
      "requests": 1000,
      "size": 320
    }
  },
  "status": 0
}
```

**Comment by Lewisyixin** [comment]
又进行了一些测试，发现是这个upstream nodes里的scheme导致的，如果改成https就会出现"全局路由错乱"的情况，改回http并且reload apisix之后全局恢复正常

**Reply by tokers** [reply]
@Lewisyixin This is a known bug and was fixed. What's your APISIX version?

**Reply by Lewisyixin** [reply]
2.14.1

**Comment by tzssangglass** [comment]
> 所以想先了解下类似情况有没有什么能进一步自查的方法？

I have seen these types of questions in the community, but I have no way to reproduce them.

You can adjust the log level to info, and then give the full log before and after the bug occurs.

**Comment by tzssangglass** [comment]
maybe fixed by: https://github.com/apache/apisix/pull/7466

**Reply by Lewisyixin** [reply]
我上午搞了个新环境能够稳定复现成功复现问题，
创建两条路由即可，
第一条路由/r1正常代理http后端n1，第二条/r2代理给https后端n2，一旦请求/r2，再次请求/r1 /r2会导致后端转发混乱(随机)

整体看上去和这个issue比较接近  我升级下版本试试
另外问下，之前我是yum安装的，我现在重新yum同版本，能获取到已经修复问题的2.14.1版本吗？

**Reply by Lewisyixin** [reply]
看了下版本内容，应该是得需要升级到2.15才行

**Reply by tzssangglass** [reply]
`sudo yum install apisix-2.15.0`

**Reply by tzssangglass** [reply]
ping @Lewisyixin , do you verify that your bug was fixed by the `apisix-2.15.0`?

**Reply by Lewisyixin** [reply]
We just had  disscussion on https://github.com/apache/apisix/discussions/7734.
I have checked this bug is welled  fixed on 2.15.0 but I have to fix error data.
Anyway, whatever the result of this disscussios(https://github.com/apache/apisix/discussions/7734.), I think there should be a  way to remind people avoid this risk of "route invalid" since https://github.com/apache/apisix/pull/7584 will be fixed in the next version

**Comment by chenyongzhi001** [comment]
同遇到+1 感觉路由的流量往不同的上游转发了
